pkg/build/apiserver/apiserver.go

```diff
@@ -0,0 +1,172 @@
+package apiserver
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/registry/rest"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	restclient "k8s.io/client-go/rest"
+	kclientsetexternal "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	kclientsetinternal "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	kubeletclient "k8s.io/kubernetes/pkg/kubelet/client"
+
+	buildapiv1 "github.com/openshift/origin/pkg/build/apis/build/v1"
+	buildclientset "github.com/openshift/origin/pkg/build/generated/internalclientset"
+	buildgenerator "github.com/openshift/origin/pkg/build/generator"
+	buildetcd "github.com/openshift/origin/pkg/build/registry/build/etcd"
+	"github.com/openshift/origin/pkg/build/registry/buildclone"
+	buildconfigregistry "github.com/openshift/origin/pkg/build/registry/buildconfig"
+	buildconfigetcd "github.com/openshift/origin/pkg/build/registry/buildconfig/etcd"
+	"github.com/openshift/origin/pkg/build/registry/buildconfiginstantiate"
+	buildlogregistry "github.com/openshift/origin/pkg/build/registry/buildlog"
+	"github.com/openshift/origin/pkg/build/webhook"
+	"github.com/openshift/origin/pkg/build/webhook/bitbucket"
+	"github.com/openshift/origin/pkg/build/webhook/generic"
+	"github.com/openshift/origin/pkg/build/webhook/github"
+	"github.com/openshift/origin/pkg/build/webhook/gitlab"
+	imageclient "github.com/openshift/origin/pkg/image/generated/internalclientset"
+)
+
+type BuildConfig struct {
```

Comment:
s/BuildConfig/BuildServerConfig/ 
Just to avoid confusion

Reply:
> s/BuildConfig/BuildServerConfig/
> Just to avoid confusion

done